extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -146,6 +146,9 @@ public class MultiStageQueryContext
 
   public static final String CTX_IS_REINDEX = "isReindex";
 
+  public static final String CTX_MAX_NUM_SEGMENTS = "maxNumSegments";
+  static final int DEFAULT_MAX_NUM_SEGMENTS = Integer.MAX_VALUE;
```

Comment:
+1 of having this null. Which means we can entirely skip the validation since user can always set this value to Integer.MAX_VALUE . Though other things will fail if the user tries to do this but point being if the user sets Integer.MAX_VALUE, we cannot skip the check.

Reply:
Missed updating it to null here. Updated it now, thanks.